### PR TITLE
Files for pysimCoder on nucleo-h743zi2

### PR DESCRIPTION
**Pull request by robertobucher**
## Summary
These files are required to use the nucleo-h743ZI2 board with pysimCoder

The following features are possible:
5 ADC
4 DI
8 DO (including 3 LED)
2 PWM channels with complementary signals
2 encoders

## Impact
arch/arm/src/stm32h7/stm32_adc.c   (output formats)
arch/arm/src/stm32h7/stm32_qencoder.c   (output formats)
board/qrm/stm32h7/nucleo-h743zi2/include/board.h   (timers for PWM and encoders)
board/qrm/stm32h7/nucleo-h743zi2/src/Makefile   (additional files)
board/qrm/stm32h7/nucleo-h743zi2/src/nucleo-h743zi2.h (gpio)
board/qrm/stm32h7/nucleo-h743zi2/src/stm32_adc.c  (new)
board/qrm/stm32h7/nucleo-h743zi2/src/stm32_bringup.c
board/qrm/stm32h7/nucleo-h743zi2/src/stm32_gpio.c (new)
board/qrm/stm32h7/nucleo-h743zi2/src/stm32_pwm.c  (new)
board/qrm/stm32h7/nucleo-h743zi2/src/stm32_qencoder.c  (new)

## Testing
Tested using pysimCoder on nucleo-H743ZI2 board

### CODE REVIEW

File: boards/arm/stm32h7/nucleo-h743zi2/src/stm32_pwm.c [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+#include "chip.h"
+#include "arm_internal.h"
+#include "stm32_pwm.h"
+#include "nucleo-h743zi2.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define OK 0
```
Comment (xiaoxiang781216) [inline boards/arm/stm32h7/nucleo-h743zi2/src/stm32_pwm.c:43]: why not use common enum from include/sys/types.h

File: boards/arm/stm32h7/nucleo-h743zi2/include/board.h [hunk 2]
```diff
@@ -390,8 +390,14 @@
 #define GPIO_TIM1_CH2OUT  GPIO_TIM1_CH2OUT_2  /* PE11 - D5 */
 #define GPIO_TIM1_CH2NOUT GPIO_TIM1_CH2NOUT_3 /* PE10 - D40 */
 
+#define GPIO_TIM1_CH1IN GPIO_TIM1_CH1IN_2       /* PE9    */
```
Comment (xiaoxiang781216) [inline boards/arm/stm32h7/nucleo-h743zi2/include/board.h:393]: why not merge into the second patch

Comment (robertobucher) [reply boards/arm/stm32h7/nucleo-h743zi2/include/board.h:393]: You're right! 
I'll remove the OK definition

**Comment by xiaoxiang781216** [comment]
@robertobucher please squash your patch into the reasonable set by:
git rebase --interactive HEAD~5
and change pick to squash

File: boards/arm/stm32h7/nucleo-h743zi2/src/stm32_adc.c [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+ * Private Data
+ ****************************************************************************/
+
+#ifdef CONFIG_STM32H7_ADC1
+/* Identifying number of each ADC channel: Variable Resistor.
+ *
+ * ADC1: {5, 10, 12, 13, 15};
+ */
+
+static const uint8_t  g_adc1_chanlist[ADC1_NCHANNELS] =
```
Comment (pkarashchenko) [inline boards/arm/stm32h7/nucleo-h743zi2/src/stm32_adc.c:70]:
```suggestion
static const uint8_t g_adc1_chanlist[ADC1_NCHANNELS] =
```

File: boards/arm/stm32h7/nucleo-h743zi2/src/stm32_adc.c [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+};
+#endif
+
+#ifdef CONFIG_STM32H7_ADC3
+/* Identifying number of each ADC channel: Variable Resistor.
+ *
+ * ADC3: {6,};
+ */
+
+static const uint8_t  g_adc3_chanlist[ADC1_NCHANNELS] =
```
Comment (pkarashchenko) [inline boards/arm/stm32h7/nucleo-h743zi2/src/stm32_adc.c:98]:
```suggestion
static const uint8_t g_adc3_chanlist[ADC1_NCHANNELS] =
```

File: boards/arm/stm32h7/nucleo-h743zi2/src/stm32_qencoder.c [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+ *
+ * Description:
+ *   All STM32H7 architectures must provide the following interface to work
+ *   with examples/qencoder.
+ *
+ ****************************************************************************/
+
+int stm32_qencoder_initialize(const char *devpath, int timer)
+{
+  int ret = 0;
```
Comment (pkarashchenko) [inline boards/arm/stm32h7/nucleo-h743zi2/src/stm32_qencoder.c:53]:
```suggestion
  int ret;
```

**Pull request closed without merging**